### PR TITLE
DateRangeBox: add new API description

### PR DESCRIPTION
**Pull request by sergepilipchuk**

### CODE REVIEW

**Review by Zedwag — APPROVED** [review]

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/Configuration.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+##### shortDescription
+An object defining configuration properties for the {WidgetName} UI component.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/Configuration.md]:
```suggestion
An object that defines configuration properties for the {WidgetName} UI component.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/applyButtonText.md [hunk 2]
```diff
@@ -5,7 +5,6 @@ default: 'OK'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies the text displayed on the Apply button.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/applyButtonText.md]:
```suggestion
Specifies the Apply button's text.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/calendarOptions.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+id: DateBoxBase.Options.calendarOptions
+type: dxCalendar_Options
+default: {}
+---
+---
+##### shortDescription
+Configures the calendar's value picker. Applies only if the [pickerType](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#pickerType) is *"calendar"*.
+
+---
+You can specify most of the [Calendar properties](/Documentation/ApiReference/UI_Components/dxCalendar/Configuration/) in this object except those listed below. These properties are overridden by their {WidgetName} counterparts.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/calendarOptions.md]:
```suggestion
You can specify the [Calendar properties](/Documentation/ApiReference/UI_Components/dxCalendar/Configuration/) in this object except those listed below. These properties are overridden by their {WidgetName} counterparts.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/cancelButtonText.md [hunk 4]
```diff
@@ -5,7 +5,6 @@ default: 'Cancel'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies the text displayed on the Cancel button.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/cancelButtonText.md]:
```suggestion
Specifies the Cancel button's text.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/dateOutOfRangeMessage.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+id: DateBoxBase.Options.dateOutOfRangeMessage
+type: String
+default: 'Value is out of range'
+---
+---
+##### shortDescription
+Specifies the message displayed if the specified date is later than the [max]({basewidgetpath}/Configuration/#max) value or earlier than the [min]({basewidgetpath}/Configuration/#min) value.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/dateOutOfRangeMessage.md]:
```suggestion
Specifies the message displayed if the date is after the [max]({basewidgetpath}/Configuration/#max) value or before the [min]({basewidgetpath}/Configuration/#min) value.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+id: DateBoxBase.Options.dateSerializationFormat
+type: String
+default: undefined
+---
+---
+##### shortDescription
+Specifies the date-time value serialization format. Use it only if you do not specify the [value](/Documentation/ApiReference/UI_Components/dxCalendar/Configuration/#value) at design time.
+
+---
+Without a value, the UI component cannot detect its format. In this case, specify the **dateSerializationFormat** property that supports the following formats:
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md]:
```suggestion
Without a value, the UI component cannot detect the format. In this case, specify the **dateSerializationFormat** property that supports the following formats:
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+- `"yyyy-MM-dd"` - a local date
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md]: ` a local date ` Any local date? How is this different from the local date and time below?

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+---
+Without a value, the UI component cannot detect its format. In this case, specify the **dateSerializationFormat** property that supports the following formats:
+
+- `"yyyy-MM-dd"` - a local date  
+
+- `"yyyy-MM-ddTHH:mm:ss"` - local date and time  
+
+- `"yyyy-MM-ddTHH:mm:ssZ"` - the UTC date and time  
+
+- `"yyyy-MM-ddTHH:mm:ssx"` - date and time with a timezone
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md]:
```suggestion
- `"yyyy-MM-ddTHH:mm:ssx"` - date and time with the timezone
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/disabledDates.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
+---
+id: DateBoxBase.Options.disabledDates
+type: Array<Date> | function(data)
+default: null
+---
+---
+##### shortDescription
+Specifies dates that users cannot select. Applies only if [pickerType](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#pickerType) is *"calendar"*.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/disabledDates.md:8]: `cannot select` Cannot select what?

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/disabledDates.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+
+##### param(data): Object
+Information about the checked date.
+
+##### field(data.component): dxDateBox
+The UI component's [instance](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Methods/#instance).
+
+##### field(data.date): Date
+The currently checked date.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/disabledDates.md]:
```suggestion
The checked date.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/displayFormat.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
+---
+id: DateBoxBase.Options.displayFormat
+type: Format
+default: null
+---
+---
+##### shortDescription
+Specifies the date display format. Ignored if the [pickerType](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#pickerType) property is *"native"*.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/displayFormat.md]:
```suggestion
Specifies the date's display format. Ignored if the [pickerType](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#pickerType) property is *"native"*.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/displayFormat.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+---
+Set [useMaskBehavior](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#useMaskBehavior) to **true** to control user input according to the display format.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/displayFormat.md]:
```suggestion
Set [useMaskBehavior](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#useMaskBehavior) to **true** to limit user input according to the display format.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/min.md [hunk 13]
```diff
@@ -5,7 +5,6 @@ default: undefined
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+The minimum date that can be selected within the UI component.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/min.md]:
```suggestion
The earliest date that can be selected in the UI component.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/max.md [hunk 14]
```diff
@@ -5,7 +5,6 @@ default: undefined
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+The last date that can be selected within the UI component.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/max.md]:
```suggestion
The latest date that can be selected in the UI component.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/invalidDateMessage.md [hunk 15]
```diff
@@ -5,7 +5,6 @@ default: 'Value must be a date or time'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies a message displayed if the typed value is not a valid date or time.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/invalidDateMessage.md]:
```suggestion
Specifies a message that is displayed if the typed value is not a valid date or time.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/todayButtonText.md [hunk 16]
```diff
@@ -5,7 +5,6 @@ default: 'Today'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specified the text displayed on the Today button.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/todayButtonText.md]:
```suggestion
Specified the Today button's text.
```

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/useMaskBehavior.md [hunk 17]
```diff
@@ -0,0 +1,10 @@
+---
+id: DateBoxBase.Options.useMaskBehavior
+type: Boolean
+default: false
+---
+---
+##### shortDescription
+Specifies whether to control user input with a mask created based on the [displayFormat](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#displayFormat) property.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/DateBoxBase/1 Configuration/useMaskBehavior.md]:
```suggestion
Specifies whether to limit user input with a mask created based on the [displayFormat](/Documentation/ApiReference/UI_Components/dx{WidgetName}/Configuration/#displayFormat) property.
```

File: api-reference/10 UI Components/dxDateBox/1 Configuration/text.md [hunk 18]
```diff
@@ -4,7 +4,7 @@ type: String
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+The read-only property that holds the text displayed by the UI component **input** element.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/dxDateBox/1 Configuration/text.md]:
```suggestion
The read-only property that stores the text displayed by the UI component **input** element.
```

File: api-reference/10 UI Components/dxDateRangeBox/1 Configuration/Configuration.md [hunk 19]
```diff
@@ -0,0 +1,7 @@
+---
+##### shortDescription
+An object defining configuration properties for the DateRangeBox UI component.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/dxDateRangeBox/1 Configuration/Configuration.md]:
```suggestion
An object that defines configuration properties for the DateRangeBox UI component.
```

File: api-reference/10 UI Components/dxDateRangeBox/1 Configuration/endDate.md [hunk 20]
```diff
@@ -0,0 +1,11 @@
+---
+id: dxDateRangeBox.Options.endDate
+type: Date | Number | String
+default: null
+---
+---
+##### shortDescription
+Specifies the end date of range.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/dxDateRangeBox/1 Configuration/endDate.md]:
```suggestion
Specifies the range's end date.
```

File: api-reference/10 UI Components/dxDateRangeBox/1 Configuration/endDateLabel.md [hunk 21]
```diff
@@ -5,7 +5,6 @@ default: 'End Date'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies a label of the end date input field.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Components/dxDateRangeBox/1 Configuration/endDateLabel.md]:
```suggestion
Specifies the label of the end date input field.
```

**Review by dirk-pieterse — APPROVED** [review]

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md (hunk 7 above)
Comment (sergepilipchuk) [reply api-reference/10 UI Components/DateBoxBase/1 Configuration/dateSerializationFormat.md]: I'm going to remove this information at all.

File: api-reference/10 UI Components/DateBoxBase/1 Configuration/disabledDates.md (hunk 9 above)
Comment (sergepilipchuk) [reply api-reference/10 UI Components/DateBoxBase/1 Configuration/disabledDates.md:8]: Specifies **dates** that users cannot select. It is about **dates**!

**Pull request merged**